src/patterns/passwords/index.md

```diff
@@ -69,6 +69,12 @@ To help users meet your password constraints and prevent mistyped passwords, you
 - show the last typed character of their password
 - make them enter their password twice and automatically compare them
 
+#### Showing and hiding passwords
+One common method for helping users type passwords is a show/hide button near the password input. When the password is hidden, the button will change the input styling to 'show' the password. When the password is visible, the button will 'hide' the password again.
+
+When there are two or more password fields on a page, the 'show' and 'hide' labels for each password input must be different.
```

Comment:
Just stumbled across this and had a thought: 

My understanding is that the reason many forms have two separate password fields is to ensure that the user has typed the password correctly, as they are unable to see it.

But by adding a show password toggle you no longer need to have two fields, as the user can validate the password is typed correctly by viewing it in plain text instead. 

Do we actually need to provide guidance for multiple show/hide password toggles? Our guidance should probably push against having multiple password fields in the first place, as they are [redundant entry](https://www.w3.org/WAI/WCAG22/Understanding/redundant-entry.html).

Reply:
I'll just chime in on the question about relevancy to 'redundant entry' to note that there's an exception carved out specifically for security. Typing a password and reducing keystroke errors in that password through redundant entry seems like a good use of that exception.

As far as I understand it. the intent of 'confirm password' inputs is to ensure that the person doesn't mis-type one of the entries. Since many people may type without carefully reading the password, even if it's visible on the screen, it'll be valuable even if they can visually see the password characters. Additionally, some people may be in an environment where showing sensitive information like a password or password confirmation on-screen is a security risk. So it's still best to have the hidden characters as the default.